SonarQube.TeamBuild.PreProcessor/SonarLintAnalyzerProvider.cs

```diff
@@ -0,0 +1,139 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SonarLintAnalyzerProvider.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using SonarQube.Common;
+using SonarQube.TeamBuild.Integration;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace SonarQube.TeamBuild.PreProcessor
+{
+    /// <summary>
+    /// Handles fetch the SonarLint ruleset and provisioning the SonarLint assemblies on the client
+    /// </summary>
+    /// <remarks>This code is a short-term solution introduced in MSBuild Scanner version 1.1.
+    /// Going forwards, we want to provide a more general solution that will work for all Roslyn
+    /// analyzers that the SonarQube server knows about. It is likely that the general-purpose
+    /// solution will be quite different.</remarks>
```

Comment:
We already have an RulesetGenerator class that handles fetching FxCop rules. Rather than amending that class I decided to create a new class that handles all of the SonarLint interactions i.e. getting the ruleset and the binaries.